src/ferui-components/tree-view/tree-view-component.ts

```diff
@@ -0,0 +1,399 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostBinding,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  QueryList,
+  ViewChild,
+  ViewChildren,
+} from '@angular/core';
+import {
+  TreeViewColorTheme,
+  TreeViewConfiguration,
+  TreeViewEvent,
+  TreeViewEventType,
+  TreeNodeData,
+  TreeNodeDataRetriever,
+  PagedTreeNodeDataRetriever,
+  TreeNode,
+  FuiTreeViewComponentStyles,
+  NonRootTreeNode,
+} from './interfaces';
+import { FuiVirtualScrollerComponent } from '../virtual-scroller/virtual-scroller';
+import { Observable, fromEvent, Subscription } from 'rxjs';
+import { throttleTime } from 'rxjs/internal/operators';
+import { FuiTreeNodeComponent } from './tree-node-component';
+
+@Component({
+  selector: 'fui-tree-view',
+  template: `
```

Comment:
Please use changeDetection: ChangeDetectionStrategy.OnPush and run the changes manually through changeDetector.markForCheck();.